src/staking-v3/hooks/useAprV3.ts

```diff
@@ -0,0 +1,103 @@
+import { $api } from 'boot/api';
+import { ethers } from 'ethers';
+import { ref, watch } from 'vue';
+import { EraLengths } from '../logic';
+import { useDappStaking } from './useDappStaking';
+
+interface InflationParam {
+  maxInflationRate: string;
+  adjustableStakersPart: string;
+  baseStakersPart: string;
+  idealStakingRate: string;
+  bonusPart: string;
+}
+
+export const useAprV3 = () => {
+  const stakerApr = ref<number>(0);
+  const bonusApr = ref<number>(0);
+  const { eraLengths, isVotingPeriod } = useDappStaking();
+
+  const percentageToNumber = (percent: string): number => {
+    return parseFloat(String(percent)) * 0.01;
+  };
+
+  const toAstr = (wei: string): number => {
+    return Number(ethers.utils.formatEther(BigInt(wei)));
+  };
+
+  const getCyclePerYear = (eraLength: EraLengths): number => {
```

Comment:
Can you explain cycle

Reply:
It's at B11 in the Dino's spreadsheet.

https://docs.astar.network/docs/learn/tokenomics2/Inflation#cycles